### PR TITLE
Change logging library to debug.

### PR DESCRIPTION
**Pull request by paulirish**
ref #428 

we dont need most of npmlog. With this change our dep tree simplifies a bunch.

### CODE REVIEW

**Comment by paullewis** [comment]
You'll need to add coverage for https://github.com/GoogleChrome/lighthouse/blob/master/lighthouse-core/core.js#L62-L63 as well

**Comment by brendankenny** [comment]
does this mean we need to always use the two argument form of `log`? It seems weird for the simple `log.log('whatever')` case for the message to be the title or whatever

File: lighthouse-core/lib/log.js [hunk 1]
```diff
@@ -16,9 +16,39 @@
...
+  if (!loggers[title]) {
+    loggers[title] = debug(title);
+  }
+  return loggers[title].apply(null, args);
+}
+
+module.exports = {
+  setLevel,
+  log: function(title) {
+    return _log(title, arguments);
```
Comment (brendankenny) [inline lighthouse-core/lib/log.js]: ugh, we really need to upgrade to node 6 and rest parameters
```
log: function(title, ...args) {
  return _log(title, args);
}
```
and avoid all the slice business above

File: lighthouse-core/lib/log.js [hunk 2]
```diff
@@ -16,9 +16,39 @@
...
+    debug.enable('*:error');
+  } else {
+    debug.enable('*, -*:verbose');
+  }
+}
 
-module.exports = isNode ? npmlog : console;
+let loggers = {};
+function _log(title, logargs) {
+  const args = Array.from(logargs).slice(1);
```
Comment (brendankenny) [inline lighthouse-core/lib/log.js]: can also do `[...logargs].slice(1)`, though it doesn't matter much

File: lighthouse-core/lib/log.js [hunk 3]
```diff
@@ -16,9 +16,39 @@
...
+  if (!loggers[title]) {
+    loggers[title] = debug(title);
+  }
+  return loggers[title].apply(null, args);
```
Comment (brendankenny) [inline lighthouse-core/lib/log.js]: `return loggers[title](...args);`

**Comment by brendankenny** [comment]
the `debug` dependency is small, but I'm still not totally seeing the point of it. You could just shut off the level-specific log levels since you're already routing them into the private `_log`. Is it the color per `title`?

Other than that, a question and a nit above

**Comment by brendankenny** [comment]
closing in favor of #481

**Comment by paulirish** [comment]
cheers. thanks!

**Pull request closed without merging**